cl/scrapers/transformer_extractor_utils.py

```diff
@@ -0,0 +1,47 @@
+import json
+from tempfile import NamedTemporaryFile
+
+import requests
+from django.conf import settings
+from django.core import serializers
+
+
+def serialize_audio_object(af):
+    """Convert serialize audio object into json for processing.
+
+    :param af: Audio File object
+    :return: Modified serailized audio file for processing in BTE
+    """
+    af_dict = json.loads(serializers.serialize("json", [af]))[0]["fields"]
+    docket_dict = json.loads(serializers.serialize("json", [af.docket]))[0][
+        "fields"
+    ]
+    court_dict = json.loads(serializers.serialize("json", [af.docket.court]))[
+        0
+    ]["fields"]
+    af_dict["docket"] = docket_dict
+    af_dict["docket"]["court"] = court_dict
+    with NamedTemporaryFile(suffix=".json") as tmp:
+        with open(tmp.name, "w") as json_data:
+            json.dump(af_dict, json_data)
+        with open(tmp.name, "rb") as file:
+            af = file.read()
```

Comment:
Why not just use `dumps` to convert something from an in-memory object to the str you want? Unless I'm missing something, I think your instincts failed you badly here. This code should have triggered a big red flag when you wrote it. (Didn't we have a similar experience with `loads` not too long ago?)

I also don't think all the (confusing and brittle) work you're doing to serialize the object is worth it. You're sending the whole object (docket, court and all) to BTE and that obfuscates the fields you actually need. I'm guessing you only really need a few of them (plus the binary data), and that you could more easily just send those. If you do that, it's more clear which fields matter and you have a clean API both here and in BTE. With a narrower API, BTE doesn't have to be designed for some weirdo format determined by Django's serializer and can just have an API for the fields it needs.

Finally, have you seen the [`json` argument for `request.post`](https://stackoverflow.com/a/26344315/64911)? I didn't know it existed, but when I saw your code, my instincts told me it probably did, and I went and found it. I think using the `json` param would avoid even using the `dumps` method above.

Reply:
@mlissner thanks for this feedback, Ive taken as second look and stripped out the unfortunately brittle code here and restructured the serialization in a much more readable and smart way (atleast I hope so).  

As for ```json``` in requests post, I could be wrong, but I don't believe you can post a file w/ the json argument in requests.  Instead I think you have one of two options, convert the json to a file and post two files (also not ideal) or send the json as parameters with the file.